c++/src/kj/string.h

```diff
@@ -721,15 +721,13 @@ inline constexpr ArrayPtr<const char> StringPtr::asArray() const {
 }
 
 inline bool StringPtr::operator==(const StringPtr& other) const {
-  return content.size() == other.content.size() &&
-      memcmp(content.begin(), other.content.begin(), content.size() - 1) == 0;
+  // The -1 is to ignore the NUL character
+  return content.first(content.size() - 1) == other.content.first(other.content.size() - 1);
 }
 
-inline bool StringPtr::operator<(const StringPtr& other) const {
-  bool shorter = content.size() < other.content.size();
-  int cmp = memcmp(content.begin(), other.content.begin(),
-                   shorter ? content.size() : other.content.size());
-  return cmp < 0 || (cmp == 0 && shorter);
+inline bool StringPtr::operator< (const StringPtr& other) const {
+  // This does not include the minus one as operator== so that it short circuts on the '\0' char in the memcmp
```

Comment:
I doubt this makes a difference, and in any case it wasn't my intent in the original code, so I'd remove the comment. Otherwise it implies that someone has actually measured this and found this to be an important optimization which is misleading.

Reply:
Comment removed.

Reply:
There's some very simple benchmarking built into kj-test, see `TestCase::doBenchmark()`.